### PR TITLE
[CI] Enables support for pytorch ci build in ARC + introduces _linux-build-rg.yml.

### PR DESCRIPTION
**Pull request by jeanschmidt**
Introduce changes related to enable ARC to run on build for linux-jammy-py3.8-gcc11

Depends on:
* https://github.com/pytorch/pytorch/pull/121908
* https://github.com/pytorch/pytorch/pull/121907
* Force docker to update credentials: https://github.com/pytorch/test-infra/pull/4991
* Add permissions to role to access ECR: https://github.com/pytorch/ci-infra/commit/acc0154aa0289369411c2f3105a21ce6010a1da4
* Add permissions to the role to access relevant S3 bucket: https://github.com/pytorch/ci-infra/commit/496b0422c360fb5ca98b9de5db0e1d8f749b22cc

## Reasoning for introducing a new `_linux-build-rg.yml`

Old style `runs-on` definition accept a string, new style `runs-on` requires a object in the format:

```
--- old
... 
  runs-on: "linux.2xlarge"
...
--- new
...
  runs-on:
    group: "running-group"
...
```

In other words, to specify a group the format of the yaml needs to be changed. Unfortunately, there is no way to accomplish this change using any trick in the book that I am aware of. This is due to the fact that GH actions yaml are not templatable and support minimal functions / replacements. A few examples that did not work: 
* [`e234f25` (#119544)](https://github.com/pytorch/pytorch/pull/119544/commits/e234f25ba1dca63989582e521779579d9e1760ad#diff-b317d4da565a9e329ccf67e669c2ff1f4d4bc5fb0ffa4d74132545ad66f84339R76)
* [`087de4a` (#119544)](https://github.com/pytorch/pytorch/pull/119544/commits/087de4ad8ba5babc95334d0d65d17293adf7b660#diff-b317d4da565a9e329ccf67e669c2ff1f4d4bc5fb0ffa4d74132545ad66f84339R76)
* [`f03512e` (#119544)](https://github.com/pytorch/pytorch/pull/119544/commits/f03512e344bf46d7d0cd5687061a4a482ee1e9de#diff-b317d4da565a9e329ccf67e669c2ff1f4d4bc5fb0ffa4d74132545ad66f84339R76)
* [`67581fb` (#119544)](https://github.com/pytorch/pytorch/pull/119544/commits/67581fb737d6f2482d53fd344e4fa91ebdb8a029#diff-b317d4da565a9e329ccf67e669c2ff1f4d4bc5fb0ffa4d74132545ad66f84339R76)

### CODE REVIEW

File: .ci/pytorch/build.sh [hunk 1]
```diff
@@ -355,3 +358,5 @@ if [[ "$BUILD_ENVIRONMENT" != *libtorch* && "$BUILD_ENVIRONMENT" != *bazel* ]];
 fi
 
 print_sccache_stats
+
+sudo chown -R $WORKSPACE_ORIGINAL_OWNER_ID /var/lib/jenkins/workspace
```
Comment (jeanschmidt) [inline .ci/pytorch/build.sh:362]: There is a linting warning here, I'll fix once get greens to avoid loosing the test state

**Review by seemethere — APPROVED** [review]

**Comment by jeanschmidt** [comment]
@huydhn does reverting this change solved inductor?

**Comment by malfet** [comment]
- IMO `[BE]` prefix is very wrong for this kind of PR
- Adding new `action` requires some soak time as it will break every currently open PR, can we phase it somehow
- Should we open CI sev to ask everyone to rebase

**Comment by jeanschmidt** [comment]
Keeping `_linux-build.yml` and maintaining the pointer to this file as for now. In a few days we'll be changing to point to `_linux-build-label.yml` and eventually to `_linux-build-rg.yml`;

Tested `_linux-build-rg.yml` again on https://github.com/pytorch/pytorch/pull/122322

**Pull request closed without merging**